segment.go

```diff
@@ -38,8 +38,18 @@ func init() {
 	reflectStaticSizeSegmentBase = int(unsafe.Sizeof(sb))
 }
 
+// Open returns a zap impl of a segment along with keeping some config values in
+// mind during the segment's lifetime.
```

Comment:
Nit: I think the "in mind" can be removed since we're dealing with segments here :)